app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -97,7 +97,7 @@ def queue_name_for_ems(ems)
       return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?
```

Comment:
I would suggest that the return ems below is a bug.  Can you find any code expecting/handling a non-string return value?  

My suggestion is:

host.rb:
```ruby
def queue_name
  # note, self is only because of that strange conditional below: return ems unless ems.kind_of (ExtManagementSystem)
  acts_as_ems? ? "generic" : self
end
```

per_ems_worker_mixin.rb
```ruby
def queue_name
  self # for that weird conditional, remove this if that is a bug
end

def queue_name_for_ems(ems)
  ems.queue_name
end
```

ext_management_system.rb

```ruby
def queue_name
  "ems_#{ems.id}"
end
```

@durandom @agrare do you have any idea what the host queue_name is for?  Is it still needed?